src/macos/liquid_glass.rs

```diff
@@ -1,28 +1,89 @@
 use std::{ffi::c_void, ptr::NonNull};
 
-use objc2::MainThreadMarker;
+use objc2::{msg_send, rc::Retained, runtime::AnyObject, sel, MainThreadMarker};
 use objc2_app_kit::{
-    NSAppKitVersionNumber, NSAutoresizingMaskOptions, NSColor, NSGlassEffectViewStyle, NSView,
-    NSWindowOrderingMode,
+    NSAppKitVersionNumber, NSAutoresizingMaskOptions, NSBox, NSColor, NSGlassEffectViewStyle,
+    NSView, NSWindowOrderingMode,
 };
-use objc2_foundation::NSInteger;
+use objc2_foundation::{NSInteger, NSRect};
 
 use crate::{macos::ns_glass_effect_view_tagged::NSGlassEffectViewTagged, Error};
 
 /// NSView::tag for NSVisualEffectViewTagged, just a random number
 pub const NS_VIEW_TAG_GLASS_VIEW: NSInteger = 96945937;
 
+const MOVED_CONTENT_KEY: &str = "WindowVibrancyMovedContentKey";
+const BACKGROUND_VIEW_KEY: &str = "WindowVibrancyBackgroundViewKey";
+
+extern "C" {
+    fn objc_setAssociatedObject(
+        object: *const c_void,
+        key: *const c_void,
+        value: *const c_void,
+        policy: usize,
+    );
+    fn objc_getAssociatedObject(object: *const c_void, key: *const c_void) -> *mut c_void;
+}
```

Comment:
do these fns work for us? https://docs.rs/objc2/latest/objc2/?search=associated

Reply:
also tbh that associated object stuff is a bit above my head. i don't really get what it's doing here

Reply:
Got it ( I think😂) thanks!